keras/utils/np_utils.py

```diff
@@ -16,12 +16,33 @@ def to_categorical(y, nb_classes=None):
     # Returns
         A binary matrix representation of the input.
     '''
-    y = np.array(y, dtype='int')
+    y = np.asarray(y, dtype='int')
     if not nb_classes:
         nb_classes = np.max(y)+1
     Y = np.zeros((len(y), nb_classes))
-    for i in range(len(y)):
-        Y[i, y[i]] = 1.
+    Y[np.arange(len(y)), y] = 1.
+    return Y
+
+
+def to_smoothed_categorical(y, nb_classes=None, smooth_factor=0):
```

Comment:
Why implement two things in one? Why not a "smooth_array" utility function (or something of the sort)?